vendor/views/vendor.py

```diff
@@ -369,12 +361,12 @@ def get_context_data(self, *args, **kwargs):
         return context
 
 
-class SubscriptionsListView(LoginRequiredMixin, ListView):
+class SubscriptionsListView(LoginRequiredMixin, SetSiteToRequestMixin, ListView):
     model = Receipt
     template_name = 'vendor/purchase_list.html'
 
     def get_queryset(self):
-        receipts = self.request.user.customer_profile.get(site=get_current_site(self.request)).receipts.filter(status__gte=PurchaseStatus.COMPLETE)
+        receipts = self.request.user.customer_profile.get(site=self.request.site).receipts.filter(status__gte=PurchaseStatus.COMPLETE)
```

Comment:
use the new function here to get the site object instead of modifying the request object.

Reply:
it should be something like this:
`receipts = self.request.user.customer_profile.get(site= get_site_from_request (self.request)).receipts.filter(status__gte=PurchaseStatus.COMPLETE)`

Reply:
On this also, is it guaranteed that there will be a customer_profile at this point?  It could throw an exception in chaining the call if it's not.

Reply:
changed to:
```
   def get_queryset(self):
        try:
            receipts = self.request.user.customer_profile.get(site=get_site_from_request(self.request)).receipts.filter(status__gte=PurchaseStatus.COMPLETE)
        except ObjectDoesNotExist:
            raise Http_404(_("Not Found"))
        subscriptions = [ receipt for receipt in receipts.all() if receipt.order_item.offer.terms > TermType.PERPETUAL and receipt.order_item.offer.terms < TermType.ONE_TIME_USE ]
        return subscriptions
```